systemd/systemd.go

```diff
@@ -1379,7 +1379,7 @@ Options={{join .Options ","}}
 LazyUnmount=yes
 
 [Install]
-WantedBy=default.target multi-user.target
+WantedBy=snapd.mounts.target
```

Comment:
it seems we don't strictly need to rewrite preexisting mount units? but it would be cleaner to do so, we have discussed but never implemented code to do that, the closest code in spirit would be ServiceManager.ensureSnapServicesUpdated, it's unclear that the same temporal placement in the new running snapd is appropriate for mount unit though.

Reply:
I have opened #12076 as a separate PR.